src/Microsoft.DotNet.Build.Tasks/BinClashLogger.cs

```diff
@@ -0,0 +1,231 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class BinClashLogger : ILogger
+    {
+        internal class ProjectState
+        {
+            public ProjectState(ProjectStartedEventArgs args)
+            {
+                GlobalProperties = args.GlobalProperties;
+                ProjectFile = args.ProjectFile;
+                ContextId = args.BuildEventContext.ProjectContextId;
+                ParentContextId = args.ParentProjectBuildEventContext.ProjectContextId;
+                TargetPath = (string)args.Properties?.Cast<DictionaryEntry>().FirstOrDefault(p => ((string)p.Key).Equals("TargetPath")).Value;
+            }
+
+            public bool RanBuild { get; set; }
+            public IDictionary<string, string> GlobalProperties { get; }
+            public string TargetPath { get; }
+            public string ProjectFile { get; }
+            public int ContextId { get; }
+            public int ParentContextId { get; }
+        }
+
+        /// <summary>
+        /// All projects evaluations that built, indexed by context ID.
+        /// </summary>
+        private Dictionary<int, ProjectState> _projectHistory = new Dictionary<int, ProjectState>();
+        private string _logFile = null;
+        private StreamWriter _fileWriter = null;
+
+        private bool _append = false;
```

Comment:
Good catch.  I added it with the intent to make it configurable but forgot to parse the param string.